test/dataset_collection/test_dataset_collection.py

```diff
@@ -18,165 +18,156 @@
 class TestDatasetCollection(TestCase):
     current_directory = Path(__file__).absolute().parent
 
-    def test_determine_type_without_return_without_arguments(self):
+    def setUp(self):
         filepath = self.current_directory / "Example.java"
-        ast = AST.build_from_javalang(build_ast(filepath))
+        self.ast = AST.build_from_javalang(build_ast(filepath))
+        self.temp_filename = self.current_directory / 'temp.java'
```

Comment:
temp file is also is unique for each test, otherwise the error will occur like 'file not found' after several file remove